vm/vm.go

```diff
@@ -191,24 +191,37 @@ func (vm *Engine) validateWitnessV0(virtualTx *wire.MsgTx, inputIdx uint32,
 		return ErrInvalidScriptVersion
 	}
 
-	// An input MUST have a prev out and also a valid witness.
-	if witness.PrevID == nil || len(witness.TxWitness) == 0 {
+	// An input must have a valid witness.
+	if len(witness.TxWitness) == 0 {
 		return newErrKind(ErrInvalidTransferWitness)
 	}
 
-	// The parameters of the new and old asset much match exactly.
-	err := matchesAssetParams(vm.newAsset, prevAsset, witness)
-	if err != nil {
-		return err
-	}
-
-	// Update the virtual transaction input with details for the specific
-	// Taproot Asset input and proceed to validate its witness.
-	virtualTxCopy := tapscript.VirtualTxWithInput(
-		virtualTx, prevAsset, inputIdx, witness.TxWitness,
+	var (
+		prevOutFetcher *txscript.CannedPrevOutputFetcher
+		err            error
 	)
 
-	prevOutFetcher, err := tapscript.InputPrevOutFetcher(*prevAsset)
+	// Genesis grouped assets will have a nil PrevID and match the prevAsset
+	// since it is a copy of the original asset. The prevOut used will need
+	// to built from the group key and not the script key.
+	switch vm.newAsset.HasGenesisWitnessForGroup() {
+	case true:
+		prevOutFetcher, err = asset.GenesisPrevOutFetcher(*prevAsset)
+
+	case false:
+		// An input MUST have a prev out and also a valid witness.
+		if witness.PrevID == nil {
+			return newErrKind(ErrInvalidTransferWitness)
+		}
+
+		// The parameters of the new and old asset much match exactly.
+		err := matchesAssetParams(vm.newAsset, prevAsset, witness)
```

Comment:
This overwrites the outer `err`, making the comparison below incorrect (this is what the linter is complaining about).